woocommerce/woocommerce-ios
# [Tracks] - Add accessibility settings to device info properties.

Closes: #7575 

- [ ] ⚠️ Please review https://github.com/Automattic/Automattic-Tracks-iOS/pull/214 ⚠️
- [ ] ⚠️ @selanthiraiyan update Podfile after https://github.com/Automattic/Automattic-Tracks-iOS/pull/214 is merged and released ⚠️

### Description
To track iOS device accessibility settings, we have added the "preferred content size category" and is accessibility "category properties" to the `deviceProperties` of all tracks events.

Internal ref - pe5sF9-xl-p2

### Testing instructions

⚠️ Please share if you know of a better way of testing the actual events in Tracks. 😄 

1. Add a breakpoint to [this line](https://github.com/Automattic/Automattic-Tracks-iOS/blob/wcios/track-accessibility-font-setting/Sources/Event%20Logging/TracksEventService.m#L55).
1.  Change the iOS device accessibility font setting by navigating to "Settings -> Accessibility -> Display & Text Size -> Larger Text"
1. Launch the app.
1. Check the `deviceProperties` value from the breakpoint and observe that `device_info_preferred_content_size_category` and `device_info_is_accessibility_category` values reflect the accessibility setting. 
1.  Change the device accessibility font settings again. 
1. Open the app and navigate to a screen if needed to catch the breakpoint again. 
1. Check the `deviceProperties` value from the breakpoint and observe that `device_info_preferred_content_size_category` and `device_info_is_accessibility_category` values reflect the latest accessibility setting. 

### Screenshots
<img width="1222" alt="Screenshot 2022-08-26 at 3 56 39 PM" src="https://user-images.githubusercontent.com/524475/186884498-707635af-cb26-4a93-90a8-c0cafd702411.png">

https://user-images.githubusercontent.com/524475/186886919-5f4e7cc5-157c-4f8a-92c5-f02f842ff8bc.mov



---
